core/core.go

```diff
@@ -250,6 +255,28 @@ func initPrivateKey(path string) (p2pcrypto.PrivKey, error) {
 	return nil, err
 }
 
+func initNetworkID(networkID int, meshDB *meshdb.MeshDB) error {
+	metadata, err := meshDB.GetMetadata()
+	if _, ok := err.(db.NotFoundError); ok {
+		// No stored metadata found (first startup)
+		if err := meshDB.SaveMetadata(networkID); err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
```

Comment:
Slight preference for checking if `err != nil` first, and then nesting the err type checking beneath it. We should really only do the error type checking if we have an error.

```
if err != nil {
  	if _, ok := err.(db.NotFoundError); ok {
		// No stored metadata found (first startup)
		if err := meshDB.SaveMetadata(networkID); err != nil {
			return err
		}
		return nil
	}
        return err
}
```